### ISSUE TITLE
Version 2.5.0 Beta: Unable to switch to BJCP 2021 styles (error 500)

### ISSUE THREAD

**Issue by grmelacz** [geoffhumphrey/brewcompetitiononlineentry#1331]
**BCOE&M Info**
Version: 2.5.0 Beta
Installation URL: https://soutez.mhmeeting.cz
Hosted Installation? No

**Describe the Bug**
I'm unable to convert the styles from BJCP 2015 to 2021 in the website prefs.

**To Reproduce**
1. Go to Website Preferences
2. Select BJCP 2021 styleset in the dropdown menu
3. Submit the settings screen
4. A blank page with HTTP 500 status is shown on URL /includes/process.inc.php?section=admin&action=edit&dbTable=preferences&id=15. 
5. The styleset is not updated.

**Expected Behavior**
The preferred styleset is set to BJCP 2021.

**Screenshots**
<img width="979" alt="Screenshot 2022-08-17 at 19 28 27" src="https://user-images.githubusercontent.com/41264018/185204657-953a4677-acfd-4ac5-ac2a-ac2cc278325e.png">

Log entry that may or may not be associated with this issue:
`
AH01071: Got error 'PHP message: PHP Warning:  Base directory /<BCOE&M directory>/classes/htmlpurifier/standalone/HTMLPurifier/DefinitionCache/Serializer does not exist,\\n                    please create or change using %Cache.SerializerPath in /<BCOE&M directory>/classes/htmlpurifier/HTMLPurifier.standalone.php on line 15688PHP message: PHP Warning:  Base directory 
`

**Server Environment**
- PHP Version: 8.0.1
- MySQL Version: 5.6.17
- Other Info: Apache

**Local Environment**
 - OS: Mac OS
 - Browser: Safari / Brave
 - Browser Version: Safari Version 15.6 (17613.3.9.1.5) / Brave Version 1.41.100 Chromium: 103.0.5060.134 (Official Build) (arm64)

**Additional Context**
Any other settings made in the same screen around the dropdown are saved correctly, only this one is not.

**Comment by geoffhumphrey** [comment]
Hi @grmelacz - thanks for reporting. I'm having trouble recreating the issue. One thing that does stand out to me is this line:

> A blank page with HTTP 500 status is shown on URL /includes/process.inc.php?section=admin&action=edit&dbTable=preferences&id=15.

Specifically, the id=15 at the end. This should be id=1 (there should only be one row in the preferences db table). Is the id of your preferences row 15?

In any case, I'll look into that why the id is incorrect.

**Comment by grmelacz** [comment]
Well, this a bit more fun than I though. Suddenly, I discovered, the styles were in the end converted. Maybe something cached somewhere. Not sure. The app does however work as desired now.

No, the prefs row has id 1.

This bug may be probably associated with my problematic upgrade. I wanted to upgrade from 2.3.something to 2.4.0 and the upgrade failed somewehere in the begining:
```
PHP Fatal error:  Array and string offset access syntax with curly braces is no longer supported in /data/web/virtuals/161777/virtual/www/subdom/soutez/classes/capitalize_name/parser.php on line 457
```
But I did not know that at the time of upgrade (logs had to be enabled on request & to get more info). I did not have the time to dig into that so I decided to upgrade straight to 2.5.0 Beta and that went smoothly.
